app/controllers/broadcast_announcements_controller.rb

```diff
@@ -0,0 +1,54 @@
+class BroadcastAnnouncementsController < ApplicationController
+  before_action :set_broadcast_announcement, only: %i[edit update destroy]
+
+  def index
+    @broadcast_announcements = BroadcastAnnouncement.all.select { |announcement| announcement.organization_id == current_organization.id }
+  end
+
+  def new
+    @broadcast_announcement = BroadcastAnnouncement.new
+  end
+
+  def edit
+  end
+
+  def create
+    @broadcast_announcement = BroadcastAnnouncement.new(broadcast_announcement_params)
+
+    respond_to do |format|
+      if @broadcast_announcement.save
+        format.html { redirect_to broadcast_announcements_url, notice: "Broadcast announcement was successfully created." }
+      else
+        format.html { render :new, status: :unprocessable_entity }
```

Comment:
Are we swallowing the errors? They should be displayed to the user somehow (e.g. via an `alert` flash message).

Reply:
I think I might be misunderstanding you, is there another error that wouldn't show up?
Right now the form errors reload the :new page and show up like this:
![image](https://user-images.githubusercontent.com/94578504/223864018-4c544559-51ec-485b-82fd-c27b29594cac.png)

I fixed most of the things, I'll have more time this weekend to finish with this and the specs

Reply:
My bad, looks good here :)